Add support for more file extensions

rodrigo2019 opened the pull request:
*Add support for more image file extensions compatible with OpenCV
*Removed validExts from list_files, so in that way the function can find all files
*Removed "replace(" ", "\\ ")", files with spaces is no more affected

jrosebr1 commented:
Thank you for the PR @rodrigo2019, I appreciate it. However, I do not think it's a good idea to include more default `validExts`. Depending on which OpenCV extensions you have installed some of these files may not automatically work.

Additionally, `validExts` is modifiable via `list_files` so I would assume a user could call that function instead, no?

> Removed "replace(" ", "\ ")", files with spaces is no more affected

Did you double check this? I'm not sure that's true although I could be wrong.

rodrigo2019 commented:
Hello Adrian, my motivation to change this code was because of the problem with file names that have spaces, I did not do understand why it is necessary to use "replace(" ", "\\ ") ", I just removed and the worked function as a charm, I'm using python 3.5.3 on windows.
If this solution doesn't work on linux or with python 2.7, I can find another way to do that, I think it is important to add support for these type of file names

After that I saw that list_files and list_images do almost the same thing. But if you don't read the code, and just get the name of function "list_files" is explicity to me that this function find all kind of files, and "list_images" all kind of images. So I changed list_files more in a generic way and list_images in a specific way.

jrosebr1 commented:
Thank you for the clarification @rodrigo2019. I appreciate it. We should test this in Linux + Python 2.7 as well to ensure it works.

As for the `list_files` and `list_images` function, I suggest we keep the original set of image extensions for the image I/O library reasons mentioned above. However, let's add `validExts` as a keyword argument the `list_images` function so the user can supply their own set of image paths as well.

rodrigo2019 commented:
@jrosebr1  I tested the function with Python2 and Python3 on Windows10 and Ubunto 16.04
The test was a simple scripts that list all files in demo_images, I also added a Folder and a image with spaces in the file name.
Here you can see  all tests:
![image](https://user-images.githubusercontent.com/8016520/38473460-c171d714-3b66-11e8-987a-77dcaa072b34.png)

jrosebr1 commented:
Thank you for testing the function on the different Python versions and operating systems, @rodrigo2019. I really appreciate your contribution. I'm going to do tests on my own system with the code from PyImageSearch posts and then merge the PR.
